drivers/adc/ad713x/ad713x.c

```diff
@@ -0,0 +1,561 @@
+	}
+	if (dev->gpio_dclkio) {
+		ret = gpio_remove(dev->gpio_dclkmode);
+		if(IS_ERR_VALUE(ret))
+			return FAILURE;
+	}
+	if (dev->gpio_mode) {
+		ret = gpio_remove(dev->gpio_mode);
+		if(IS_ERR_VALUE(ret))
+			return FAILURE;
+	}
+	if (dev->gpio_pnd) {
+		ret = gpio_remove(dev->gpio_pnd);
+		if(IS_ERR_VALUE(ret))
+			return FAILURE;
+	}
+	if (dev->gpio_resetn) {
+		ret = gpio_remove(dev->gpio_resetn);
+		if(IS_ERR_VALUE(ret))
+			return FAILURE;
+	}
+
+	return SUCCESS;
+}
+
+/**
+ * @brief Initialize the wideband filter bandwidth for every channel.
+ *        ad713x_init() helper function.
+ * @param [in] dev - AD713X device handler.
+ * @return \ref SUCCESS in case of success, \ref FAILURE otherwise.
+ */
+static int32_t ad713x_init_chan_bw(struct ad713x_dev *dev)
+{
+	int8_t i;
+	int32_t ret;
+
+	for (i = CH3; i >= 0; i--) {
+		ret = ad713x_wideband_bw_sel(dev, i, 0);
+		if (IS_ERR_VALUE(ret))
+			break;
+	}
+
+	return SUCCESS;
+}
+
+/**
+ * @brief Initialize the device.
+ * @param device - The device structure.
+ * @param init_param - The structure that contains the device initial
+ *                     parameters.
+ * @return \ref SUCCESS in case of success, \ref FAILURE otherwise.
+ */
+int32_t ad713x_init(struct ad713x_dev **device,
+		    struct ad713x_init_param *init_param)
+{
+	struct ad713x_dev *dev;
+	int32_t ret;
+	uint8_t data;
+
+	dev = (struct ad713x_dev *)calloc(1, sizeof(*dev));
+	if (!dev)
+		return FAILURE;
+
+	if (!init_param->spi_common_dev) {
+		ret = spi_init(&dev->spi_desc, &init_param->spi_init_prm);
+		if (IS_ERR_VALUE(ret))
+			goto error_dev;
+	} else {
+		dev->spi_desc = calloc(1, sizeof *dev->spi_desc);
+		dev->spi_desc->chip_select = init_param->spi_init_prm.chip_select;
+		dev->spi_desc->extra = init_param->spi_common_dev->extra;
+		dev->spi_desc->max_speed_hz = init_param->spi_init_prm.max_speed_hz;
+		dev->spi_desc->mode = init_param->spi_init_prm.mode;
+	}
+
+	ret = ad713x_init_gpio(dev, init_param);
+	if(IS_ERR_VALUE(ret))
+		goto error_gpio;
+	/*
+	 * Tie this pin to IOVDD for master mode operation,
+	 * tie this pin to IOGND for slave mode operation.
+	 */
+	gpio_direction_output(dev->gpio_mode, init_param->mode_master_nslave);
+	/* Tie this pin low to ground to make DLCK operating in gated mode */
+	gpio_direction_output(dev->gpio_dclkmode, init_param->dclkmode_free_ngated);
+	/*
+	 * Tie this pin high to make DCLK an output,
+	 * tie this pin low to make DLCK an input.
+	 */
+	gpio_direction_output(dev->gpio_dclkio, init_param->dclkio_out_nin);
+
+	/* Get the ADCs out of power down state */
+	gpio_direction_output(dev->gpio_pnd, init_param->pnd);
+
+	/* Reset to configure pins */
+	gpio_direction_output(dev->gpio_resetn, false);
+	mdelay(100);
+	gpio_set_value(dev->gpio_resetn, true);
+
+	mdelay(100);
+
+	dev->dev_id = init_param->dev_id;
+
+	ret = ad713x_spi_reg_read(dev, AD713X_REG_DEVICE_CONFIG, &data);
+	if (IS_ERR_VALUE(ret))
+		goto error_gpio;
+	data |= AD713X_DEV_CONFIG_PWR_MODE(HIGH_POWER);
```

Comment:
nitpick: for cosmetic reasons i would add some extra lines between these it makes it easier to follow. The same i would do for other places in the code where things are very tight together and a bit harder to follow where one ends.